src/pages/home/report/ReportActionsView.js

```diff
@@ -252,7 +252,17 @@ class ReportActionsView extends React.Component {
     updateSortedReportActions(reportActions) {
         this.sortedReportActions = _.chain(reportActions)
             .sortBy('sequenceNumber')
-            .filter(action => action.actionName === 'ADDCOMMENT' || action.actionName === 'IOU')
+            .filter((action) => {
+                // Only show non-empty ADDCOMMENT actions or IOU actions
+                const message = _.first(lodashGet(action, 'message', null));
+                const html = lodashGet(message, 'html', '');
+                const text = lodashGet(message, 'text', '');
+                return action.actionName === 'IOU'
+                    || (
+                        action.actionName === 'ADDCOMMENT'
+                        && (html !== '' || text !== '')
```

Comment:
I don't think we need to check for both text and HTML in this code. There is discussion elsewhere that `text` is going away. Since `html` is all that is stored in the DB, I think this logic should be improved to only look at `html`.

Reply:
perfect, happy to remove it, it just makes it more complex to read